shell/platform/android/android_platform_surface_gl.cc

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/android/android_platform_surface_gl.h"
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <GLES/gl.h>
+#include <GLES/glext.h>
+#include <jni.h>
+#include "flutter/common/threads.h"
+#include "flutter/lib/ui/painting/resource_context.h"
+#include "flutter/shell/platform/android/platform_view_android_jni.h"
+#include "third_party/skia/include/core/SkSurface.h"
+#include "third_party/skia/include/gpu/GrBackendSurface.h"
+#include "third_party/skia/include/gpu/GrTexture.h"
+#include "third_party/skia/include/gpu/GrTypes.h"
+
+namespace shell {
+
+AndroidPlatformSurfaceGL::~AndroidPlatformSurfaceGL() {
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::IO()->PostTask([this, &latch]() {
+    glDeleteTextures(1, &texture_id_);
+    latch.Signal();
+  });
+  latch.Wait();
+}
+
+AndroidPlatformSurfaceGL::AndroidPlatformSurfaceGL() {
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::IO()->PostTask([this, &latch]() {
+    GrGLuint texID;
```

Comment:
I bet you can generate this lazily in `AndroidPlatformSurfaceGL::MakeSkImage`. Better yet, since I don't this there is actually any texture upload after the bind, you can create a new texture, bind, make the update texuture image call and put the texture collection routine in the `TextureReleaseProc` (which we know will be collected on the GPU thread). No more thread hopping. As a general rule, never call `blink::Threads::IO()->PostTask`. To see why, refer to the issues I raised in the iOS variant of `MakeSkImage`.